### PR TITLE
userspace: multi-desk SPA support

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
- Refactors file-server heavily, stripping out serving files from eyre, and simplifying the necessary codepaths. As a result, all assets for SPAs should be included in the glob. This actually greatly simplifies the process of making an SPA grid-compatible, as all that is needed is to change the basename on the router, build and then glob the resultant `dist/` folder. Additionally, it should be marginally faster, as %file-server no longer has to scry out occasionally.
- Introduces %docket, an agent for managing the fetching and updating of SPA globs. %docket reads one or many .docket files at the root of desks, parsing out the URL to fetch the glob and a path to bind the resultant glob to.
- Fixes miscellaneous breakage caused in userspace by the move to software distribution.
- Introduces a subscription path that emits facts whenever a desk is installed or updated. %docket uses this to discover dockets.
- Adds a mark for %vats so that grid can dynamically generate a list of installed desks.

TODO:
- [ ] Delete %glob, move generation to a thread
- [ ] Cleanup %file-server datastructures
- [ ] Add %btc-wallet, webterm docket
- [ ] Be more conservative about refetching globs

### CODE REVIEW

File: pkg/arvo/app/spider.hoon [hunk 1]
```diff
@@ -524,7 +524,7 @@
   =+    .^
       =tube:clay
       %cc
-      /(scot %p our.bowl)/[q.byk.bowl]/(scot %da now.bowl)/[output]/json
+      /(scot %p our.bowl)/landscape/(scot %da now.bowl)/[output]/json
```
[the remark this answers is not in the document]
Comment (tacryt-socryp) [reply pkg/arvo/app/spider.hoon:527]: ???

File: pkg/arvo/app/spider.hoon [hunk 2]
```diff
@@ -364,11 +364,11 @@
     ==
   =/  pax=path
     ~|  no-file-for-thread+file
-    (need (get-fit:clay beak %ted file))
+    (need (get-fit:clay beak(r da+now.bowl) %ted file))
   :_  state
   :_  ~
   :+  %pass  /build/[new-tid]
-  [%arvo %c %warp p.beak q.beak ~ %sing %a r.beak pax]
```
[the remark this answers is not in the document]
Comment (tacryt-socryp) [reply pkg/arvo/app/spider.hoon:371]: ???

File: pkg/landscape/app/docket.hoon [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+      |-  
+      ?~  dockets  [cards state]
+      =^  crds  state
+        (add-docket:cc i.dockets)
+      $(cards (welp crds cards), dockets t.dockets)
+    ::
+    ++  remove
+      |=  updates=(set [=desk name=cord])
+      ^-  (quip card _state)
+      `state
```
Comment (tacryt-socryp) [inline pkg/landscape/app/docket.hoon:180]: This looks like it hasn't been implemented yet. Just flagging

File: pkg/landscape/app/glob.hoon [hunk 4]
```diff
@@ -167,6 +156,16 @@
       (on-load !>(new-state))
     (on-poke:def mark vase)
   ==
+  ::
+  ++  to-mime
+    =/  home=path  /(scot %p our.bowl)/work/(scot %da now.bowl)
```
Comment (tacryt-socryp) [inline pkg/landscape/app/glob.hoon:161]: Is %work a new desk?

**Comment by liam-fitzgerald** [comment]
closing in favour of the more recent PR https://github.com/urbit/urbit/pull/5154

**Pull request closed without merging**